### PR TITLE
[LANGUAGE] Allowing negative numbers on calculations and comparisons.

### LINKED ISSUE #3072: [BUG] Can't use negative numbers
**Describe the bug**
We can't use negative numbers in computations: assign it to a variable, use it in conditionals or in expressions.

**Paste the Hedy code & level**
When using in a conditional it says that the dash is not allowed.
```
antwoord is 0
if antwoord is -10
  antwoord is ask 'A number not between -10 and 10'
print 'Goed gedaan!
```
image

When assigning it to a variable it throws Missing Command Exception:
image

**Expected behavior**
I think kids should be able to work with negative numbers! We allow them to use floating points numbers, which I think is more complex to understand than negative numbers!

### PR DESCRIPTION
**Pull request by jpelay**
**Description**

Changes several parts of the grammars to allow for negative numbers inside several parts of the grammar:

* Changes the rule NUMBER to be able to accept both negative and positive numbers.
* Adds numbers as one of the possible parsing options when making equality checks.
* Allows to use negative numbers inside of expressions such as: -4+5.

**Fixes #3072**

**How to test**

I added a few test across several levels.

### CODE REVIEW

**Comment by boryanagoncharenko** [comment]
This will clash with the PR for refactoring of level 12 tests (https://github.com/Felienne/hedy/pull/3108). Since that PR introduces a rather big change, is it possible to review it first? If it is accepted and merged to main, I will resolve the conflicts of this PR, so that it can also be merged to main.

**Comment by jpelay** [comment]
> This will clash with the PR for refactoring of level 12 tests (https://github.com/Felienne/hedy/pull/3108). Since that PR introduces a rather big change, is it possible to review it first? If it is accepted and merged to main, I will resolve the conflicts of this PR, so that it can also be merged to main.

Yes, no problem!

**Comment by Felienne** [comment]
> > This will clash with the PR for refactoring of level 12 tests (#3108). Since that PR introduces a rather big change, is it possible to review it first? If it is accepted and merged to main, I will resolve the conflicts of this PR, so that it can also be merged to main.
> 
> Yes, I have no problem!

Are you going to review #3108 @jpelay or do you want me to?

**Comment by jpelay** [comment]
> > > This will clash with the PR for refactoring of level 12 tests (#3108). Since that PR introduces a rather big change, is it possible to review it first? If it is accepted and merged to main, I will resolve the conflicts of this PR, so that it can also be merged to main.
> > 
> > 
> > Yes, I have no problem!
> 
> Are you going to review #3108 @jpelay or do you want me to?

I can review it!

File: grammars/level1.lark [hunk 1]
```diff
@@ -33,7 +33,7 @@ command: print | ask | echo | turtle | error_invalid_space | error_invalid
 print: _PRINT (text)?
 ask: _ASK (text)?
 echo: _ECHO (text)?
-turtle: _FORWARD ((NEGATIVE_NUMBER | NUMBER | text))? -> forward | _TURN ((left | right ))? -> turn | _COLOR ((black | blue | brown | gray | green | orange | pink | purple | red | white | yellow | text))? -> color
+turtle: _FORWARD ((NUMBER | text))? -> forward | _TURN ((left | right ))? -> turn | _COLOR ((black | blue | brown | gray | green | orange | pink | purple | red | white | yellow | text))? -> color
```
Comment (Felienne) [inline grammars/level1.lark:36]: Ah lot more readable!!

**Review by Felienne — APPROVED** [review]
Well done @jpelay, adding a feature and also simplifying the grammar <3

**Pull request merged**